### DISCUSSION TITLE
Use the web_google_maps widget at portal fields

### DISCUSSION THREAD

**Discussion by macagua** [gityopie/odoo-addons#167] [General]
@gityopie  Hi I am using the ``web_google_maps`` add-on in the Odoo 13 CE version, 

I need to adapt the view of ``portal_my_details`` of the ``portal`` add-on, when the website user enters the company name (aka the ``company_name`` field) or the address street (aka the ``street`` field), then I need overrirde these fields with the information from Google Maps using the Widget that support your addons.

So I have created a module that has a template that inherits from ``portal.portal_my_details`` to customize the field ``company_name``, this module installs correctly, but I notice that it does not work correctly in the Odoo frontend in the view ``<template id="portal_my_details">`` in the route ``/my/account/`` because when I edit the address of the website user, autocompletion is not activated.

Is it because the ``web_google_maps`` assets files do not load them on the website? If this is how I would do to register in my module that it loads the assets BUT on the Odoo frontend, because in the Odoo backend they load perfectly, what do you recommend me to do?

**Comment by gityopie** [comment]
Hi @macagua 
Yes the JS code for backend and frontend are two different things, they cannot be shared.
To make it work on website you need to develop a module. You can check how I done it on version 12 (https://github.com/gityopie/odoo-addons/tree/12.0/website_google_address_form) but I guess, you need to adapt the code to match the new way Odoo interact with DOM element (widget for website), check this example https://github.com/odoo/odoo/blob/13.0/addons/website/static/src/js/user_custom_javascript.js

**Reply by macagua** [reply]
@gityopie Thanks for answering, now I have published my improvements to your addons in the following repository https://github.com/macagua/odoo-addons-google-maps 

I am not a jQuery / Javascript programmer, my main skell in Odoo is backend but if you have an example of how this Javascript code is migrated, I can try to migrate it to version 13.

**Comment by gityopie** [comment]
@macagua I've pushed a new branch https://github.com/gityopie/odoo-addons/tree/13.0-website-gautocomplete. On this branch you can find a new module `website_portal_google_address_form`. 
Please have a look and let me know your feedback
